lib/protocol/protocol.go

```diff
@@ -641,9 +641,9 @@ func (c *rawConnection) handleResponse(resp Response) {
 	c.awaitingMut.Unlock()
 }
 
-func (c *rawConnection) send(msg message, done chan struct{}) bool {
+func (c *rawConnection) send(msg message, done chan struct{}) (sent bool) {
 	defer func() {
-		if done != nil {
+		if !sent && done != nil {
```

Comment:
Why not do the if & close just before the "return true" below, it's just one place and avoids the defer entirely?